spec/realtime/auth.test.js

```diff
@@ -71,8 +72,9 @@ define(['ably', 'shared_helper', 'async'], function(Ably, helper, async) {
 			}
 
 			var authPath = echoServer + "/?type=json&body=" + encodeURIComponent(JSON.stringify(tokenDetails));
-
-			realtime = helper.AblyRealtime({ authUrl: authPath });
+			var clientOptions = new ClientOptions();
+			clientOptions.authUrl = authPath;
```

Comment:
If you have introduced a new `ClientOptions` class into JS, should we not have a set of tests to ensure that the class has the default values?  

Also, if we are taking this approach now, are we not creating inconsistencies unless we provide constructors now for all the types, see https://docs.ably.io/realtime/types/?